HollowEngine/BulletHell/src/DungeonGeneration/DungeonFloor.cpp

```diff
@@ -0,0 +1,344 @@
+        }
+        if (printMode & 2)
+        {
+            for (int i = 0; i < mHeight; i++)
+            {
+                for (int j = 0; j < mWidth; j++)
+                {
+                    const DungeonRoom& room = mRooms[Index(i, j)];
+                    DungeonRoomType roomType = room.mRoomType;
+                    std::cout << room.mRoomID << " ";
+
+                }
+                std::cout << std::endl;
+            }
+        }
+        if (printMode & 4)
+        {
+            for (int i = 0; i < mHeight; i++)
+            {
+                for (int j = 0; j < mWidth; j++)
+                {
+                    const DungeonRoom& room = mRooms[Index(i, j)];
+                    int dist = room.mDistFromEntrance;
+                    std::cout << dist << " ";
+                }
+                std::cout << std::endl;
+            }
+            std::cout << std::endl;
+        }
+        if (printMode & 8)
+        {
+            for (int i = 0; i < mHeight; i++)
+            {
+                for (int j = 0; j < mWidth; j++)
+                {
+                    const DungeonRoom& room = mRooms[Index(i, j)];
+                    int totalDoors = room.TotalDoors();
+                    std::cout << totalDoors << " ";
+                }
+                std::cout << std::endl;
+            }
+            std::cout << std::endl;
+        }
+        if (printMode & 16) // dead-ends
+        {
+            for (int i = 0; i < mHeight; i++)
+            {
+                for (int j = 0; j < mWidth; j++)
+                {
+                    const DungeonRoom& room = mRooms[Index(i, j)];
+                    if (room.mRoomType == DungeonRoomType::ENTRANCE)
+                        std::cout << "& ";
+                    else if (room.TotalDoors() == 1)
+                        std::cout << "e ";
+                    else
+                        std::cout << "  ";
+                }
+                std::cout << std::endl;
+            }
+            std::cout << std::endl;
+        }
+    }
+
+    int DungeonFloor::Index(int row, int col) const
+    {
+        return row * mWidth + col;
+    }
+
+    void DungeonFloor::Index2D(int index1D, int& row, int& col)
+    {
+        row = index1D / mWidth;
+        col = index1D % mWidth;
+    }
+
+    int DungeonFloor::SumDistance(std::vector<int> roomIndices)
+    {
+        int sum = 0;
+        for (int i : roomIndices)
+        {
+            sum += mRooms[i].mDistFromEntrance;
+        }
+        return sum;
+    }
+
+    void DungeonFloor::UpdateDoors(int roomIndex)
+    {
+        int row, col;
+        Index2D(roomIndex, row, col);
+
+        int upIndex = (row > 0) ? Index(row - 1, col) : -1;
+        int rightIndex = (col < mWidth - 1) ? Index(row, col + 1) : -1;
+        int downIndex = (row < mHeight - 1) ? Index(row + 1, col) : -1;
+        int leftIndex = (col > 0) ? Index(row, col - 1) : -1;
+
+        //up
+        if (upIndex >= 0
```

Comment:
Turn these bad boys into functions as well, looks like its doing the same thing for each direction